readingDataPointsFilters:

  - SurfaceNormalDataPointsFilter:
      knn: 5
      keepNormals: 1
      keepDensities: 1
  - ObservationDirectionDataPointsFilter
  - OrientNormalsDataPointsFilter 

referenceDataPointsFilters:
  - SurfaceNormalDataPointsFilter:
      knn: 10
      epsilon: 3.16
      keepNormals: 1
      keepDensities: 1

matcher:
  KDTreeMatcher:
    maxDist: 1.0
    knn: 10
    epsilon: 3.16

outlierFilters:
  - TrimmedDistOutlierFilter:
     ratio: 0.85
  - SurfaceNormalOutlierFilter:
     maxAngle: 0.42

errorMinimizer:
  PointToPlaneErrorMinimizer

transformationCheckers:
  - DifferentialTransformationChecker:
      minDiffRotErr: 0.001
      minDiffTransErr: 0.01
      smoothLength: 4
  - CounterTransformationChecker:
      maxIterationCount: 40


# matcher:
#   KDTreeMatcher:
#     knn: 1

# outlierFilters:
#   - MaxDistOutlierFilter:
#       maxDist: 1

# errorMinimizer:
#   PointToPointErrorMinimizer

# transformationCheckers:
#   - CounterTransformationChecker:
#       maxIterationCount: 10
#   - DifferentialTransformationChecker:
#       minDiffRotErr: 0.001
#       minDiffTransErr: 0.01
#       smoothLength: 4 


inspector:
 NullInspector
 
logger:
  FileLogger

  
